modules/@apostrophecms/ui/ui/apos/components/AposFile.vue

```diff
@@ -0,0 +1,207 @@
+<template>
+  <div>
+    <label
+      class="apos-input-wrapper apos-file-dropzone"
+      :class="{
+        'apos-file-dropzone--dragover': dragging,
+        'apos-is-disabled': disabled || limitReached
+      }"
+      @drop.prevent="uploadFile"
+      @dragover="dragHandler"
+      @dragleave="dragging = false"
+    >
+      <p class="apos-file-instructions">
+        <template v-if="dragging">
+          <cloud-upload-icon :size="38" />
+        </template>
+        <AposSpinner v-else-if="uploading" />
+        <template v-else>
+          <paperclip-icon :size="14" class="apos-file-icon" />
+          {{ messages.primary }}&nbsp;
+          <span class="apos-file-highlight" v-if="messages.highlighted">
+            {{ messages.highlighted }}
+          </span>
+        </template>
+      </p>
+      <input
+        type="file"
+        class="apos-sr-only"
+        :disabled="disabled || limitReached"
+        @input="uploadFile"
+        :accept="allowedExtensions"
+      >
+    </label>
+    <div v-if="filesOrAttachment.length" class="apos-file-files">
+      <AposSlatList
+        :value="filesOrAttachment"
+        @input="update"
+        :disabled="readOnly"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    uploading: {
+      type: Boolean,
+      default: false
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    },
+    limit: {
+      type: Number,
+      default: 1
+    },
+    attachment: {
+      type: Object,
+      default: null
+    },
+    allowedExtensions: {
+      type: String,
+      default: '*'
+    },
+    readOnly: {
+      type: Boolean,
+      default: false
+    },
+    def: {
+      type: String,
+      default: null
+    }
+  },
+  emits: [ 'upload-file', 'update' ],
+  data () {
+    return {
+      selectedFiles: [],
+      dragging: false
+    };
+  },
+  computed: {
+    limitReached () {
+      return this.filesOrAttachment.length >= this.limit;
+    },
+    filesOrAttachment () {
+      if (!this.selectedFiles.length && !this.attachment) {
+        return [];
+      }
+
+      return this.selectedFiles.length
+        ? this.selectedFiles
+        : [ this.attachment ];
+    },
+    messages () {
+      const msgs = {
+        primary: 'Drop a file here or',
+        highlighted: 'click to open the file explorer'
+      };
+      if (this.disabled) {
+        msgs.primary = 'Field is disabled';
+        msgs.highlighted = '';
+      }
+      if (this.limitReached) {
+        msgs.primary = 'Attachment limit reached';
+        msgs.highlighted = '';
+      }
+      return msgs;
+    }
+  },
+  methods: {
+    uploadFile ({ target, dataTransfer }) {
+      this.dragging = false;
+      const extensionRegex = /(?:\.([^.]+))?$/;
+
+      const files = target.files ? target.files : (dataTransfer.files || []);
+      const filteredFiles = Array.from(files).filter((file, i) => {
```

Comment:
Ok, this part was unclear since it was passing `field.accept` to the input directly but in another hand, an array of allowedExtensions was allowing all extensions ad was checked from another method. To make it clearer and less scattered i'll use the string only and use this one in the `checkFileGroup` method. `groupInfo.extensions` was empty everytime I checked.

Reply:
Also accept is not enough since it doesn't block bad type files from drag and drop.